### PR TITLE
Throw SDK exceptions so Unity dev can handle them

### PR DESCRIPTION
**Pull request by 0xFirekeeper**
This allows devs like me to handle SDK errors on the UI as they wish.

### CODE REVIEW

File: Assets/Thirdweb/Scripts/Bridge.cs [hunk 1]
```diff
@@ -90,17 +98,15 @@ public static async Task<T> InvokeRoute<T>(string route, string[] body)
...
-            try 
-            {
-                string result = await task.Task;
-                // Debug.LogFormat("Result from {0}: {1}", route, result);
-                return JsonConvert.DeserializeObject<Result<T>>(result).result;
-            } 
-            catch (Exception)
+            string result = await task.Task;
+            Debug.Log($"InvokeRoute Result: {result}");
+            if (typeof(T) == typeof(string))
```
Comment (joaquim-verges) [inline Assets/Thirdweb/Scripts/Bridge.cs:103]: so it only returns the error if you explicitely ask for a string? when you invoke? That's doesn't seem great.

I have a better idea for this. The encapsulating Result type will have 2 implementations, one for success which boxes the actual result of type T (like now) and one for error which will contain the exception and possibly more fields like a `reason` property.

I'm going to work on this this week, will probably take over this PR

File: Assets/WebGLTemplates/Thirdweb/lib/thirdweb.js [hunk 2]
```diff
@@ -4,124 +4,107 @@ import { ThirdwebSDK } from "https://esm.sh/@thirdweb-dev/sdk?bundle";
...
-    (typeof value === "object" &&
-      value !== null &&
-      value.type === "BigNumber" &&
-      "hex" in value)
-  ) {
-    return ethers.BigNumber.from(value).toString();
-  }
-  return value;
+    // if we find a BigNumber then make it into a string (since that is safe)
+    if (
```
Comment (joaquim-verges) [inline Assets/WebGLTemplates/Thirdweb/lib/thirdweb.js:8]: are these just formatting changes? for JS files we use prettier, if you can, install it too so we don't override styles over each other.

Also what formatting tool do you use for C# code? haven't found a good one yet.

**Review by joaquim-verges — CHANGES_REQUESTED** [review]

File: Assets/Thirdweb/Scripts/Bridge.cs (hunk 1 above)
Comment (0xFirekeeper) [reply Assets/Thirdweb/Scripts/Bridge.cs:103]: Definitely needs a rework, but this is the only workaround that worked for me to not get parsing issues with the current Result type from thirdweb.js when returning a single string that has unparseable stuff in it like dots, really a pain if this isn't done OR the actual result could be serialized once again on the js side I guess?

File: Assets/WebGLTemplates/Thirdweb/lib/thirdweb.js (hunk 2 above)
Comment (0xFirekeeper) [reply Assets/WebGLTemplates/Thirdweb/lib/thirdweb.js:8]: Yep mb had whitespace changes hidden by default, I used prettier with 4 tab width and 120 line length, what settings do you use so I can set it up? For C# I use the default C# Microsoft VS Code extension, I think it uses either Razor or the default C# formatter.

**Comment by 0xFirekeeper** [comment]
The main thing here is the exception throwing for me, parsing can (and definitely should) have an improved solution to help parse weird strings. Might also indirectly solve the issues once has when passing an ABI that contains dots in it.

**Comment by 0xFirekeeper** [comment]
Will make a better PR for exceptions only

**Pull request closed without merging**